python/tvm/autotvm/tuner/ga_tuner.py

```diff
@@ -0,0 +1,125 @@
+# pylint: disable=consider-using-enumerate,invalid-name,abstract-method
+
+"""Tuner with genetic algorithm"""
+
+import numpy as np
+
+from .tuner import Tuner
+
+class GATuner(Tuner):
+    """Tuner with genetic algorithm.
+    This tuner does not have a cost model so it always run measurement on real machines.
+    This tuner expands the :code:`ConfigEntity` as gene.
+
+    Parameters
+    ----------
+    pop_size: int
+        number of genes in one generation
+    elite_num: int
+        number of elite to keep
+    mutation_prob: float
+        probability of mutation of a knob in a gene
+    """
+    def __init__(self, task, pop_size, elite_num=3, mutation_prob=0.1):
+        super(GATuner, self).__init__(task)
+
+        # algorithm configurations
+        self.pop_size = pop_size
+        self.elite_num = elite_num
+        self.mutation_prob = mutation_prob
+
+        assert elite_num <= pop_size, "number of elite must be less than population size"
+
+        # space info
+        self.space = task.config_space
+        self.n_subspace = len(task.config_space.space_map)
+        self.dim_subspaces = [len(x) for x in task.config_space.space_map.values()] + [1]
```

Comment:
I think the appended `[1]` makes the current code a bit harder to read since len(self.dim_subspaces) is not equal to self.n_subspace.

It looks like the added `[1]` is only used in _gene2index(), doesn't it?  How about adding it only in _gene2index()?